public/pages/DetectorResults/containers/AnomalyResultsTable.tsx

```diff
@@ -62,6 +67,99 @@ export function AnomalyResultsTable(props: AnomalyResultsTableProps) {
       ? props.anomalies.filter((anomaly) => anomaly.anomalyGrade > 0)
       : [];
 
+  const handleOpenDiscover = async (startTime: number, endTime: number, item: any) => {
+    try {
+      // calculate time range with 10-minute buffer on each side per customer request
+      const TEN_MINUTES_IN_MS = 10 * 60 * 1000;
+      const startISO = new Date(startTime - TEN_MINUTES_IN_MS).toISOString();
+      const endISO = new Date(endTime + TEN_MINUTES_IN_MS).toISOString();
+      
+      const basePath = `${window.location.origin}${window.location.pathname.split('/app/')[0]}`;
+      
+      const savedObjectsClient = getSavedObjectsClient();
+      
+      const indexPatternTitle = props.detectorIndex.join(',');
+      
+      // try to find an existing index pattern with this title
+      const indexPatternResponse = await savedObjectsClient.find({
+        type: 'index-pattern',
+        fields: ['title'],
+        search: `"${indexPatternTitle}"`,
+        searchFields: ['title'],
+      });
+      
+      let indexPatternId;
+      
+      if (indexPatternResponse.savedObjects.length > 0) {
+        indexPatternId = indexPatternResponse.savedObjects[0].id;
+      } else {
+        // try to create a new index pattern
+        try {
+          const newIndexPattern = await savedObjectsClient.create('index-pattern', {
+            title: indexPatternTitle,
+            timeFieldName: props.detectorTimeField,
+          });
+          
+          indexPatternId = newIndexPattern.id;
+
+          getNotifications().toasts.addSuccess(`Created new index pattern: ${indexPatternTitle}`);
```

Comment:
suggestion: utilize ToastInputFields https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/public/notifications/toasts/toasts_api.tsx#L48 to pass it it a title and id. 

this will ensure you don't have duplicate toasts on screen

Reply:
thanks for the suggestion, opened an issue about refactoring this cross the repo - https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/issues/1006